### PR TITLE
exercise-eqvbintrees

### PR DESCRIPTION
**Pull request by nkoblai**

### CODE REVIEW

File: go-tour/eqvbintrees.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+// Same determines whether the trees
+// t1 and t2 contain the same values.
+func Same(t1, t2 *tree.Tree) bool {
+	ch1 := make(chan int)
+	ch2 := make(chan int)
+
+	go Walk(t1, ch1)
+	go Walk(t2, ch2)
+
+	if t1 != nil && t2 != nil {
```
Comment (arbrix) [inline go-tour/eqvbintrees.go]: this check work only for the first value in a tree not to the rest

File: go-tour/eqvbintrees.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+		v1, ok1 := <-ch1
+		v2, ok2 := <-ch2
+
+		if v1 == v2 && ok1 == ok2 {
```
Comment (arbrix) [inline go-tour/eqvbintrees.go]: do you really need this `if`?

File: go-tour/eqvbintrees.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import (
+	"fmt"
+
+	"golang.org/x/tour/tree"
+)
+
+// Walk walks the tree t sending all values
+// from the tree to the channel ch.
+func Walk(t *tree.Tree, ch chan int) {
+	if t != nil {
```
Comment (arbrix) [inline go-tour/eqvbintrees.go]: in golang usually use another approach: returns or small logic by condition and other logic is placed after that for improving readability

**Review by arbrix — CHANGES_REQUESTED** [review]
small changes required

**Comment by nkoblai** [comment]
@arbrix fixed

**Review by arbrix — APPROVED** [review]
LGTM

**Pull request merged**